### PR TITLE
Replace old manual query with derived query

### LINKED ISSUE #357: Replace query with derived query
In `BookRepository`, we currently have

```java
    @Query("select b from Book b " + "where lower(b.title) like lower(concat('%', :filterText, '%'))")
    List<Book> search(@Param("filterText") String filterText);
```

This should be replaced with a derived query (`findByTitle`)

### PR DESCRIPTION
**Pull request by flouou**
## Summary of change

Replaced manual search query with derived query

## Related issue

Closes #357 

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [ ] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-h6vx3n8l-KN8QnO50r7QWZHgHFnFdPw)

### CODE REVIEW

**Comment by knjk04** [comment]
Thanks, @flouou!

In case you weren't already aware, you will not show as a contributor on our repository's page or on your profile. One possible reason for this is because the email attached to your commit is not the same as the email associated with your GitHub account (see [here](https://docs.github.com/en/enterprise/2.13/user/articles/why-are-my-contributions-not-showing-up-on-my-profile)).

I'll assign this PR as a draft for the time being. After resolving this, or if you don't want to fix this and wish to proceed, then feel free to request a review

**Comment by flouou** [comment]
Hey @knjk04 , thanks for pointing this out, i didn't know that! I have now added the email address to my gh-account, so everything should be fine now, right?

**Comment by knjk04** [comment]
Hey, it's all good now. Thanks! I'll review this shortly

File: src/main/java/com/karankumar/bookproject/backend/repository/BookRepository.java [hunk 1]
```diff
@@ -28,8 +28,7 @@
 import java.util.Set;
 
 public interface BookRepository extends JpaRepository<Book, Long> {
-    @Query("select b from Book b " + "where lower(b.title) like lower(concat('%', :filterText, '%'))")
-    List<Book> search(@Param("filterText") String filterText);
+    List<Book> findByTitleContainingIgnoreCase(String title);
```
Comment (amit1307) [inline src/main/java/com/karankumar/bookproject/backend/repository/BookRepository.java:31]: Could you please add a test just to make sure it works as expected?

File: src/main/java/com/karankumar/bookproject/backend/service/BookService.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ public Long count() {
         if (filterText == null || filterText.isEmpty()) {
             return bookRepository.findAll();
         }
-        return bookRepository.search(filterText);
+        return bookRepository.findByTitleContainingIgnoreCase(filterText);
```
Comment (amit1307) [inline src/main/java/com/karankumar/bookproject/backend/service/BookService.java:82]: Could you please add a test just to make sure it works as expected?

**Review by amit1307 — CHANGES_REQUESTED** [review]

**Comment by flouou** [comment]
Hey @knjk04 , as per the comments made by @amit1307 I have added a test for the newly created query in BookRepository.

**Review by knjk04 — APPROVED** [review]
Thanks @flouou! All good for me

**Comment by knjk04** [comment]
@amit1307 Let us know when you're happy by approving or suggesting further changes

**Review by amit1307 — APPROVED** [review]

**Comment by knjk04** [comment]
Thanks again, @flouou!

**Pull request merged**